modules/hal_nordic/nrfx/nrfx_glue.h

```diff
@@ -334,8 +334,56 @@ void nrfx_busy_wait(uint32_t usec_to_wait);
 				   NRFX_PPI_GROUPS_USED_BY_802154_DRV | \
 				   NRFX_PPI_GROUPS_USED_BY_MPSL)
 
-/** @brief Bitmask that defines GPIOTE channels that are reserved for use outside of the nrfx library. */
-#define NRFX_GPIOTE_CHANNELS_USED NRFX_GPIOTE_CHANNELS_USED_BY_BT_CTLR
```

Comment:
We should get rid of `NRFX_GPIOTE_CHANNELS_USED_BY_BT_CTLR` completely and instead allocate GPIOTE channels properly in the Bluetooth controller. Then, all the stuff that you are adding below will be unnecessary.

Reply:
Here's my attempt to solve the problem: #73145.

Reply:
removed BT_CTRL part since #73145 got merged.